### PR TITLE
CI/Circle CI: Prepare master for circle ci

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  Added circle ci config file
-  Synced dependent files with 3.3.5

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by funjoker** [comment]
Docker: funjoker/circle-ci:master-buildpacks-disco is a placeholder for now

**Comment by jackpoz** [comment]
trinitycore/circle-ci:master-buildpacks-disco pushed to DockerHub

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            sudo chown -Rv circleci:circleci /root
+      - run:
+          name: Checkout
+          command: |
+            /scripts/checkout.sh
+      - restore_cache:
+          keys:
+            - nopch-{{ .Branch }}-{{ .Revision }}
+            - nopch-{{ .Branch }}
+            - nopch
```
Comment (jackpoz) [inline .circleci/config.yml:63]: master should have a different key (we might change that for 3.3.5 branch too )

Comment (funjoker) [reply .circleci/config.yml:63]: i would drop the nopch key and just use 
```
            - nopch-{{ .Branch }}-{{ .Revision }}
            - nopch-{{ .Branch }}
```

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**